components/clarinet-deployments/src/lib.rs

```diff
@@ -408,6 +454,69 @@ pub async fn generate_default_deployment(
         requirements_data.append(&mut boot_contracts_asts);
     }
 
+    let mut contract_diags: HashMap<QualifiedContractIdentifier, Vec<Diagnostic>> = HashMap::new();
+    let mut asts_success = true;
+
+    // Validate custom boot contracts from override_boot_contracts_source
+    if !settings.override_boot_contracts_source.is_empty() && !simnet_remote_data {
+        for (contract_name, file_path) in &settings.override_boot_contracts_source {
+            // Only validate existing boot contracts that are being overridden
+            if !clarity_repl::repl::boot::BOOT_CONTRACTS_NAMES.contains(&contract_name.as_str()) {
+                continue;
+            }
+
+            // Load and validate the custom boot contract
+            let custom_source = match file_accessor {
+                None => {
+                    // Fallback to file system when no file_accessor is provided
+                    std::fs::read_to_string(file_path)
+                        .map_err(|e| format!("Failed to read boot contract file {file_path}: {e}"))
+                }
+                Some(file_accessor) => {
+                    let sources = file_accessor
+                        .read_files(vec![file_path.to_string()])
+                        .await
+                        .map_err(|e| {
+                            format!("Failed to read boot contract file {file_path}: {e}")
+                        })?;
+                    sources
+                        .get(file_path)
+                        .ok_or_else(|| {
+                            format!("Unable to read custom boot contract: {contract_name}")
+                        })
+                        .cloned()
+                }
+            }?;
+
+            let (epoch, clarity_version) =
+                get_boot_contract_epoch_and_clarity_version(contract_name.as_str())?;
+
+            // Create a temporary contract for validation
+            let temp_contract = ClarityContract {
+                code_source: ClarityCodeSource::ContractInMemory(custom_source),
+                deployer: ContractDeployer::Address(default_deployer_address.to_address()),
+                name: contract_name.clone(),
+                clarity_version,
+                epoch: clarity_repl::repl::Epoch::Specific(epoch),
+            };
+
+            let (_, diagnostics, ast_success) = session.interpreter.build_ast(&temp_contract);
+
+            // Collect AST diagnostics instead of returning immediately
```

Comment:
Remove. Again, only make sense in the context of this PR and is reference to a comment I made